theme.js

```diff
@@ -1,117 +1,84 @@
 /* jshint node: true */
 'use strict';
 
+var path = require('path');
+
 var Addon = require('ember-cli/lib/models/addon');
+var ThemeCore = require('./lib/theme-core');
+var TransformComponentClasses = require('./lib/htmlbars-plugins/transform-component-classes');
+var TransformUITableComponents = require('./lib/htmlbars-plugins/transform-ui-table-components');
+var funnel = require('broccoli-funnel');
 
-var path = require('path');
-var csso = require('broccoli-csso');
+var themeCore;
 
-var jsPreprocessor = require('./lib/js-preprocessor');
-var templatePreprocessor = require('./lib/template-preprocessor');
-var scssPreprocessor = require('./lib/scss-preprocessor');
+var Theme = Addon.extend({
+  themeCore: null,
+  parentTheme: null,
 
-var TransformComponentClasses = require('./lib/transform-component-classes');
-var TransformUITableComponents = require('./lib/plugins/transform-ui-table-components');
+  isDevelopingAddon: function() {
+    return true;
+  },
 
-var UITheme = Addon.extend({
   init: function() {
     this._super.init.apply(this, arguments);
-    this._originalCompileStyles = this.compileStyles;
-    this.compileStyles = function() {};
-  },
 
-  treeForStyles() {
-    return this._originalCompileStyles(this.treeGenerator(this.root + '/addon/styles'));
+    this.themeCore = themeCore || new ThemeCore();
+    themeCore = this.themeCore;
+
+    this.parentTheme = this.themeCore.parentTheme;
+    this.themeCore.parentTheme = this;
+
+    this.themeCore.register(this);
   },
 
   setupPreprocessorRegistry: function(type, registry) {
```

Comment:
why does ember-cli not just have different methods for setting up the parent / self registries?